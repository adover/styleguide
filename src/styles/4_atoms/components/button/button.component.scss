@import '../../button';
@import '../../success-icon';
@import '../../../2_tools/mixins/apply-spacing';
@import '../../../2_tools/mixins/center';
@import '../../../2_tools/mixins/focus-ring';
@import '../../../2_tools/mixins/media-query';
@import '../../../2_tools/mixins/position';

/// When hovering on Primary or Secondary buttons
/// which aren't outlines we need to make sure
/// the fill of the svg is white, regardless
@mixin svg-hover {
	&:hover,
	&:active,
	&:focus {
		::ng-deep cdx-svg-icon svg use {
			fill: white !important;
			transition: map-get($transitionSpeed, default) unquote(map-get($transitionEase, default)) fill;
		}
	}
}

:host {
	@include button-init;

	text-decoration: none !important;

	&[inline='false'] {
		@include button-block;
	}

	&[inline='true'] {
		@include button-display(inline);
	}

	&:not([fullWidth='true']) {
		@include media-query($size: mobile) {
			@include button-reset-block();
		}
	}

	&[align='left'] {
		text-align: left !important;
	}

	&[fillStyle='primary'] {
		@include button-primary;
		@include svg-hover;
	}

	&[fillStyle='flat'] {
		@include button-primary-flat($hover: false);
	}

	&[fillStyle='warn'] {
		@include button-warn;
	}

	&[fillStyle='outline'] {
		@include button-primary-outline;
	}

	&[fillStyle='primaryFlatInverse'] {
		@include button-primary-flat-inverse($hover: false);
	}

	&[fillStyle='secondary'] {
		@include button-secondary;
		@include svg-hover;
	}

	&[fillStyle='secondaryFlat'] {
		@include button-secondary-flat($hover: false);
	}

	&[fillStyle='secondaryOutline'] {
		@include button-secondary-outline;
		@include svg-hover;
	}

	&[fillStyle='tertiary'] {
		@include button-tertiary;
	}

	&[fillStyle='tertiaryFlat'] {
		@include button-tertiary-flat($hover: false);
	}

	&[data-slotstart='true'] {
		&[data-slotstartsize='extra-small'] {
			padding-left: $spacing-8;
		}

		&[data-slotstartsize='small'] {
			padding-left: $spacing-10;
		}

		&[data-slotstartsize='default'] {
			padding-left: $spacing-12;
		}

		i.button-icon--start {
			left: $spacing-2;
		}
	}

	&[data-slotend='true'] {
		&[data-slotendsize='extra-small'] {
			padding-right: $spacing-8;
		}

		&[data-slotendsize='small'] {
			padding-right: $spacing-10;
		}

		&[data-slotendsize='default'] {
			padding-right: $spacing-12;
		}

		i.button-icon--end {
			display: flex;
			right: $spacing-2;
		}
	}

	&[size='large'] {
		@include button-sizing($size: large);
	}

	&[size='small'] {
		@include button-small($align: center);
	}

	&[size='large'] {
		@include button-large($align: center);
	}

	&[fillStyle='link'] {
		@include button-primary-flat($hover: false);
		@include button-link;
		@include focus-inset;
		@include section-margin($size: default);
	}
	
	// &[marginSize='small'] {
	// 	@include side-margin($size: default, $removeHorizontal: true);
	// }
	// &[marginSize='large'] {
	// 	@include side-margin($size: large, $removeHorizontal: true);
	// }

	&:disabled,
	&[aria-disabled='true'] {
		@include button-disabled;

		&[fillStyle='flat'],
		&[fillStyle*='Flat'] {
			@include button-disabled(
				$disabled-background-color: transparent,
				$disabled-color: map-deep-get($button, colours, disabled, flat-colour)
			);
		}

		&::ng-deep {
			svg {
				use {
					fill: map-deep-get($button, colours, disabled, colour) !important;
				}
			}
		}
	}

	&[fullWidth='true'] {
		@include button-block;
	}

	.btn--footer {
		@include button-small;

		border-radius: 0px;
		border-top: map-deep-get($config, borders, divider);
	}

	i {
		position: absolute;
		top: 50%;
		transform: translateY(-50%);

		&:empty {
			display: none;
		}
	}
}
