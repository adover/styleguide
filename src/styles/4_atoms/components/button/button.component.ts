import { FocusableOption, FocusOrigin } from '@angular/cdk/a11y';
import { ButtonState } from '../../../ui-models';
import {
	ChangeDetectionStrategy,
	ChangeDetectorRef,
	Component,
	ElementRef,
	EventEmitter,
	HostBinding,
	Input,
	OnDestroy,
	Output,
	ViewChild,
} from '@angular/core';

const BUSY_LOADER_HIDE_TIME = 300;

// TODO: Migrate to using these combos
export enum ButtonFillColour {
	Primary,
	PrimaryInverse,
	Secondary,
	SecondaryInverse,
	Disabled,
	Warn,
}

export enum ButtonFillStyle {
	Fill,
	Flat,
	Outline,
}

@Component({
	selector: 'cdx-button, [cdxButton]',
	template: `
		<i class="button-icon--start" #slotStart> <ng-content select="[start]"></ng-content> </i>

		<ng-container *ngIf="!isCompleted">
			<ng-content></ng-content>
		</ng-container>

		<ng-container *ngIf="isCompleted">
			<ng-content select="[completed]"></ng-content>
		</ng-container>

		<i class="button-icon--end" #slotEnd>
			<cdx-loading *ngIf="isBusy"></cdx-loading>

			<cdx-success-icon *ngIf="isCompleted" (animationCompleteEmitter)="animationCompleted()"></cdx-success-icon>

			<ng-content select="[end]"></ng-content>
		</i>
	`,
	styleUrls: ['./button.component.scss'],
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ButtonComponent implements OnDestroy, FocusableOption {
	startTimer = 0;
	state = ButtonState.default;

	@HostBinding('style.pointer-events') get events() {
		return this.state !== ButtonState.default ? 'none' : 'auto';
	}

	constructor(public el: ElementRef, private cdr: ChangeDetectorRef) {}

	@HostBinding('attr.aria-busy')
	get isBusy() {
		return this.state === ButtonState.busy;
	}

	@HostBinding('attr.data-completed')
	get isCompleted() {
		return this.state === ButtonState.completed;
	}

	@Input() class: string;

	@Input()
	@HostBinding('attr.aria-disabled')
	disabled = false;

	@HostBinding('attr.align')
	@Input()
	align: 'left' | 'center' | 'right' = 'center';

	@HostBinding('attr.fillStyle')
	@Input()
	fillStyle:
		| 'primary'
		| 'primaryFlatInverse'
		| 'outline'
		| 'link'
		| 'flat'
		| 'warn'
		| 'secondary'
		| 'secondaryFlat'
		| 'secondaryOutline'
		| 'tertiary'
		| 'tertiaryFlat' = 'primary';

	@HostBinding('attr.size')
	@Input()
	size: 'default' | 'small' | 'large' = 'default';

	// @HostBinding('attr.marginSize')
	// @Input()
	// marginSize: 'default' | 'small' = 'default';

	@HostBinding('attr.inline')
	@Input()
	inline = false;

	@HostBinding('attr.fullWidth')
	@Input()
	fullWidth = false;

	@Input()
	@HostBinding('class.btn--footer')
	footerButton = false;

	@Output() animationCompletedEmitter: EventEmitter<any> = new EventEmitter();

	// TODO: We should automate the adding of an icon into the button if we want i
	// We would then be able to dictate the colour of it and hover states
	@ViewChild('slotStart', { read: ElementRef, static: true })
	slotStart: ElementRef;
	@ViewChild('slotEnd', { read: ElementRef, static: true })
	slotEnd: ElementRef;

	_endSlotActive: boolean;
	_startSlotActive: boolean;

	// ADD SVG SIZE
	@Input()
	@HostBinding('attr.data-slotend')
	set endSlotActive(a: boolean) {
		this._endSlotActive = a;
	}

	get endSlotActive() {
		return this._endSlotActive;
	}

	@Input()
	@HostBinding('attr.data-slotstart')
	set startSlotActive(a: boolean) {
		this._startSlotActive = a;
	}

	get startSlotActive() {
		return this._startSlotActive;
	}

	@HostBinding('attr.data-slotendsize')
	@Input()
	endSlotSize: 'extra-small' | 'small' | 'default' = 'default';

	@HostBinding('attr.data-slotstartsize')
	@Input()
	startSlotSize: 'extra-small' | 'small' | 'default' = 'default';

	@Input()
	set buttonState(buttonState: ButtonState) {
		switch (buttonState) {
			case ButtonState.busy: {
				this.setBusyState();
				break;
			}

			case ButtonState.completed: {
				this.setCompleteState();
				break;
			}

			case ButtonState.errored:
			case ButtonState.default:
			default: {
				this.setDefaultState();
				break;
			}
		}
	}

	ngOnDestroy(): void {
		return;
	}

	animationCompleted() {
		this.animationCompletedEmitter.emit();
	}

	focus(origin?: FocusOrigin): void {
		this.el.nativeElement.focus();
	}

	setCompleteState() {
		this._endSlotActive = true;
		const timeDifferenceFromStart = Date.now() - this.startTimer;
		if (timeDifferenceFromStart >= BUSY_LOADER_HIDE_TIME) {
			this.setState(ButtonState.completed);
		} else {
			const remainingTime = BUSY_LOADER_HIDE_TIME - timeDifferenceFromStart;

			setTimeout(() => {
				this.setState(ButtonState.completed);
			}, remainingTime);
		}
	}

	setBusyState() {
		this._endSlotActive = true;
		this.startTimer = Date.now();
		this.setState(ButtonState.busy);
	}

	setDefaultState() {
		this._endSlotActive = false;
		this.setState(ButtonState.default);
	}

	setState(state) {
		this.state = state;
		this.cdr.markForCheck();
	}
}
