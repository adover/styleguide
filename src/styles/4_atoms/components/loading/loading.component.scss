@import '../../loading-spinner';
@import '../../../2_tools/functions/build-animation';
@import '../../../1_settings/transitions';

:host {
	display: block;
	height: map-deep-get($config, icons, size, small);
	width: map-deep-get($config, icons, size, small);

	&[data-vertical-center='true'] {
		@include loading-init();
	}

	&[data-vertical-center='false'] {
		@include loading-init($vertical-center: false);
	}

	&[data-fill='light'] {
		rect {
			fill: white;
		}
	}

	&[data-fill='highlight'] {
		rect {
			fill: map-deep-get($config, colors, background, highlight);
		}
	}

	&[data-fill='mid'] {
		rect {
			fill: map-deep-get($config, colors, background, mid);
		}
	}

	&[data-fill='dark'] {
		rect {
			fill: map-deep-get($config, colors, background, dark);
		}
	}

	&[data-fill='green'] {
		rect {
			fill: map-deep-get($config, colors, background, green);
		}
	}
}
