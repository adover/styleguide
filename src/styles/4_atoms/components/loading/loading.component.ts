import { Component, ChangeDetectionStrategy, Input, HostBinding } from '@angular/core';

@Component({
	selector: 'cdx-loading',
	templateUrl: './loading.component.html',
	styleUrls: ['./loading.component.scss'],
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class LoadingComponent {
	@HostBinding('attr.data-fill')
	@Input()
	fill: 'dark' | 'light' | 'highlight' | 'mid' | 'green' = 'light';

	@HostBinding('attr.data-vertical-center')
	@Input()
	verticalCenter: boolean;

	@HostBinding('attr.aria-live') live = 'polite';
}
