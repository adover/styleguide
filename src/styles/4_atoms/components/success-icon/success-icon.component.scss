@import '../../success-icon';

:host {
	@include success-icon-init();
}
