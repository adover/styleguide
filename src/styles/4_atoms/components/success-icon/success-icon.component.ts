import { animate, style, transition, trigger } from '@angular/animations';
import { THEME_CONSTANTS } from '../../../1_settings/theme';
import { ChangeDetectionStrategy, Component, EventEmitter, Output, OnInit } from '@angular/core';

@Component({
	animations: [
		trigger('animate', [
			transition(':enter', [
				style({ 'clip-path': 'polygon(0 0, 0 0, 0 100%, 0 100%)' }),
				animate(
					`${THEME_CONSTANTS.transitions.default}`,
					style({ 'clip-path': 'polygon(0 0, 100% 0, 100% 100%, 0% 100%)' })
				),
			]),
		]),
	],
	selector: 'cdx-success-icon',
	template: `
		<cdx-svg-icon fill="light" size="small" name="tick" [@animate]></cdx-svg-icon>
	`,
	styleUrls: ['./success-icon.component.scss'],
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class SuccessIconComponent implements OnInit {
	constructor() {}

	transition = THEME_CONSTANTS.transitions.default;

	@Output() animationCompleteEmitter: EventEmitter<any> = new EventEmitter();

	ngOnInit(): void {
		// IE11 doesn't support clip-path so just timeout instead
		setTimeout(() => {
			this.animationComplete();
		}, parseInt(this.transition, 10));
	}

	animationComplete() {
		this.animationCompleteEmitter.emit();
	}
}
