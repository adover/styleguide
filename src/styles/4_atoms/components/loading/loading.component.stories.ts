import { color, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/angular';
import { LoadingComponent } from './loading.component';

storiesOf('Atoms|Loading', module)
	.addDecorator(withKnobs)
	.add('Loading', () => {
		const fill = 'dark';
		return {
			moduleMetadata: {
				declarations: [LoadingComponent],
			},
			template: `<div><cdx-loading [fill]="fill"></cdx-loading></div>`,
			props: {
				fill,
			},
		};
	});
