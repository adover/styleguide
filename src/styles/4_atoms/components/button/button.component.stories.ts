import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { centered } from '@storybook/addon-centered/angular';
import { boolean, select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/angular';
import { ButtonComponent } from './button.component';
import { ButtonState } from '../../../model';
import { LoadingComponent } from '../loading/loading.component';
import { SuccessIconComponent } from '../success-icon/success-icon.component';
import { SvgDefinitionsComponent } from '../svg-definitions/svg-definitions.component';
import { SvgIconComponent } from '../svg-icon/svg-icon.component';

storiesOf('Atoms|Button', module)
	.addDecorator(centered)
	.addDecorator(withKnobs)
	.addParameters({ options: { selectedPanel: 'storybook/knobs/panel' } })
	.add('Button', () => {
		const fullWidth = boolean('fullWidth', false);
		const buttonSizes = {
			default: 'default',
			large: 'large',
		};
		const fillStyles = {
			Primary: 'primary',
			'Primary Flat Inverse': 'primaryFlatInverse',
			'Primary Outline': 'outline',
			Link: 'link',
			Warn: 'warn',
			'Primary Flat': 'flat',
			Secondary: 'secondary',
			'Secondary Flat': 'secondaryFlat',
			'Secondary Outline': 'secondaryOutline',
			Tertiary: 'tertiary',
			'Tertiary Flat': 'tertiaryFlat',
		};

		const sizes = ['extra-small', 'small', 'default'];

		const state = {
			default: ButtonState.default,
			busy: ButtonState.busy,
			completed: ButtonState.completed,
			errored: ButtonState.errored,
		};

		const size = select('size', buttonSizes, 'large');
		const fillStyle = select('fillStyle', fillStyles, 'primary');
		const disabled = boolean('disabled', false);
		const iconStart = boolean('Icon at start', false);
		const iconSize = select('Icon size', sizes, 'default');
		const iconEnd = boolean('Icon at end', false);
		const buttonState = select('state', state, ButtonState.default);

		return {
			moduleMetadata: {
				declarations: [
					LoadingComponent,
					ButtonComponent,
					SuccessIconComponent,
					SvgIconComponent,
					SvgDefinitionsComponent,
				],
				imports: [BrowserAnimationsModule],
			},
			template: `
      <cdx-svg-definitions></cdx-svg-definitions>

    <button
    [fullWidth]="fullWidth"
    [size]="size"
      [disabled]="disabled"
      [endSlotActive]="iconEnd"
      [endSlotSize]="iconSize"
      [fillStyle]="fillStyle"
      [startSlotActive]="iconStart"
      [startSlotSize]="iconSize"
      cdxButton>
          <cdx-svg-icon fill="dark" name="shopping-list" start [size]="iconSize" *ngIf="iconStart"></cdx-svg-icon>
          Look at my start icon <ng-container completed>Added</ng-container>
          <cdx-svg-icon fill="dark" name="chevron-bottom" end [size]="iconSize" *ngIf="iconEnd"></cdx-svg-icon>
      </button>
`,
			props: {
				fullWidth,
				buttonState,
				size,
				fillStyle,
				disabled,
				iconStart,
				iconSize,
				iconEnd,
			},
		};
	});
