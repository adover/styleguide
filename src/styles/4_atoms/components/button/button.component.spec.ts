import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { ButtonComponent } from './button.component';
import { ButtonState } from '../../../ui-models';
import { LoadingComponent } from '../loading/loading.component';
import { SuccessIconComponent } from '../success-icon/success-icon.component';

describe('ButtonComponent', () => {
	let component: ButtonComponent;
	let fixture: ComponentFixture<ButtonComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [ButtonComponent, LoadingComponent, SuccessIconComponent],
			schemas: [CUSTOM_ELEMENTS_SCHEMA],
		}).compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(ButtonComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should render with busy loader', () => {
		fixture.detectChanges();
		component.state = ButtonState.busy;
		expect(fixture).toMatchSnapshot();
	});

	it('should render without busy loader', () => {
		component.state = ButtonState.default;
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
	it('should render as disabled', () => {
		component.disabled = true;
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
	it('should render as active', () => {
		component.disabled = false;
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

	it('should render as primary', () => {
		component.fillStyle = 'primary';
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
	it('should render as secondary', () => {
		component.fillStyle = 'secondary';
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
	it('should render as outline', () => {
		component.fillStyle = 'outline';
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
	it('should render as flat', () => {
		component.fillStyle = 'flat';
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

	it('should render as large', () => {
		component.size = 'large';
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

	it('should render as default', () => {
		component.size = 'default';
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

	it('should render full width', () => {
		component.fullWidth = true;
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

	it('set button complete state', fakeAsync(() => {
		component.setState = jest.fn();
		component.buttonState = ButtonState.busy;
		component.buttonState = ButtonState.completed;
		expect(component.setState).toHaveBeenCalledWith(ButtonState.busy);
		expect(component.setState).not.toHaveBeenCalledWith(ButtonState.completed);
		expect(component.setState).toBeCalledTimes(1);
		tick(1000);
		expect(component.setState).toHaveBeenCalledWith(ButtonState.completed);
	}));
});
