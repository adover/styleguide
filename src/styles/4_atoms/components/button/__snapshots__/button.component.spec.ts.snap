// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render as active 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as default 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as disabled 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled={[Function Boolean]}
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as flat 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as large 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as outline 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as primary 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render as secondary 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render full width 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth={[Function Boolean]}
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render with busy loader 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;

exports[`ButtonComponent should render without busy loader 1`] = `
<cdx-button
  __ngContext__={[Function LRootView]}
  align={[Function String]}
  animationCompletedEmitter={[Function EventEmitter]}
  cdr={[Function ViewRef]}
  disabled="false"
  el={[Function ElementRef]}
  endSlotSize={[Function String]}
  fillStyle={[Function String]}
  footerButton="false"
  fullWidth="false"
  inline="false"
  size={[Function String]}
  slotEnd={[Function ElementRef]}
  slotStart={[Function ElementRef]}
  startSlotSize={[Function String]}
  startTimer="0"
  state={[Function String]}
>
  <i
    class="button-icon--start"
  /><i
    class="button-icon--end"
  >
    
    
  </i>
</cdx-button>
`;
